ops/model.py

```diff
@@ -769,22 +772,25 @@ class Port:
 OpenedPort = Port  # Alias for backwards compatibility.
 
 
-class LazyMapping(Mapping[str, str], ABC):
+_LazyValueType = typing.TypeVar("_LazyValueType")
+
+
+class _GenericLazyMapping(Mapping[str, _LazyValueType], ABC):
```

Comment:
I added this in the main conversation, but repeating here in context: I would prefer to have this be `LazyMapping` and just have the generic type made concrete for `RelationDataContent` (like with `ConfigData`), but that would be a backwards-incompatible change and isn't actually required to fix the bug.